halo2_backend/src/plonk.rs

```diff
@@ -62,7 +62,11 @@ where
     /// - Otherwise: Writes an uncompressed curve element with coordinates in Montgomery form
     /// Writes a field element into raw bytes in its internal Montgomery representation,
     /// WITHOUT performing the expensive Montgomery reduction.
-    pub fn write<W: io::Write>(&self, writer: &mut W, format: SerdeFormat) -> io::Result<()> {
+    pub(crate) fn write<W: io::Write>(
```

Comment:
All the methods in this file that you turned into `pub(crate)` where originally `pub` before the frontend-backend split; so I think it's best to keep them `pub` so that users of the halo2 library can use them.  See https://github.com/privacy-scaling-explorations/halo2/blob/73408a140737d8336490452193b21f5a7a94e7de/halo2_proofs/src/plonk.rs#L79

Reply:
I thought that it is better to `pub(crate)` the methods in this file, since they are only used by inner crates & the users are only allowed to use the pub crates.
But, I think you are right.
Will roll back the `pub(crate)` to `pub`

Reply:
> I thought that it is better to `pub(crate)` the methods in this file, since they are only used by `halo2_backend` crate & the users are only allowed to use the `halo2_proofs` crate.

The idea is that users that want to use the legacy halo2 API will only import `halo2_proofs`, but note that `halo2_proofs` re-exports many types from the other crates.  If the re-exported types have `pub(crate)` methods, these methods will not be available for the user.

Also, users that are interested in the frontend-backend split API will be using the `halo2_middleware`, `halo2_frontend` and `halo2_backend` crates.  The only crate that ideally users should not import is `halo2_common`.

Reply:
I think we shouldn't have public exported methods with `#[allow(unused)]`; and the CI should still pass (it was passing before this PR for this file)

Reply:
Resolve the CI issue in https://github.com/privacy-scaling-explorations/halo2/pull/276/commits/32d42cfdc9741c2b53dc3a1d02c2f4c61a225fde

Reply:
> > I thought that it is better to `pub(crate)` the methods in this file, since they are only used by `halo2_backend` crate & the users are only allowed to use the `halo2_proofs` crate.
> 
> The idea is that users that want to use the legacy halo2 API will only import `halo2_proofs`, but note that `halo2_proofs` re-exports many types from the other crates. If the re-exported types have `pub(crate)` methods, these methods will not be available for the user.
> 
> Also, users that are interested in the frontend-backend split API will be using the `halo2_middleware`, `halo2_frontend` and `halo2_backend` crates. The only crate that ideally users should not import is `halo2_common`.

I believe you are right. @ed255 
Users/devs might want to use the types/functions from several crates.
I think we can do more corrections after https://github.com/privacy-scaling-explorations/halo2/issues/266 is completed.